syntax = "proto3";

import "google/protobuf/timestamp.proto";

message Notification {
  string id = 1;
  int64 recipient_id = 2;
  string title = 3;
  string message = 4;
  bool read = 5;
  bool sent = 6;
  google.protobuf.Timestamp created_at = 7;
}

message NotificationRequest {
  string userId = 1;
}

message NotificationResponse {
  string title = 1;
  string message = 2;
  bool isRead = 3;
}

message ListNotificationsRequest {
  string user_id = 1;
}

message ListNotificationsResponse {
  repeated Notification notifications = 1;
}

message MarkAsReadRequest {
  string notification_id = 1;
}

message MarkAsReadResponse {
  bool success = 1;
}

service Notifications {
  rpc SendNotification(Notification) returns (MarkAsReadResponse);
  rpc ListNotifications(ListNotificationsRequest) returns (ListNotificationsResponse);
  rpc MarkNotificationAsRead(MarkAsReadRequest) returns (MarkAsReadResponse);
  rpc SubscribeToNotifications(NotificationRequest) returns (stream NotificationResponse);
}
